lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1592,10 +1608,27 @@ Attribute PaddedSharedEncodingAttr::parse(AsmParser &parser, Type type) {
       remainingAttrs.push_back(attr);
     }
   }
-  if (auto ctaLayout = parseCTAAttrs(parser, remainingAttrs, order.size()))
-    return parser.getChecked<PaddedSharedEncodingAttr>(
-        parser.getContext(), intervals, paddings, order, *ctaLayout);
-  return {};
+  auto ctaLayout = parseCTAAttrs(parser, remainingAttrs, order.size());
+  if (!ctaLayout)
+    return {};
+
+  // [optional] {<attr-dict>} (linear_component)
```

Comment:
Instead of using another `{..}`, what about putting in the same `{..}` where `order` and CTA attributes are in?